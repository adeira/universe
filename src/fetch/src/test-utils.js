// @flow

/* eslint-disable import/no-extraneous-dependencies */

import { setupServer, type Server } from 'msw/node';

export { rest } from 'msw';

export const server: Server = setupServer();
