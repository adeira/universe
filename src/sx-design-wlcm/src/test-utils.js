// @flow

import { init as FbtInit, IntlVariations as FbtIntlVariations } from 'fbt';
import { render as renderWithoutProviders } from '@testing-library/react';

const initFbt = () => {
  FbtInit({
    translations: {}, // TODO
    hooks: {
      getViewerContext: () => ({
        GENDER: FbtIntlVariations.GENDER_UNKNOWN,
        regionalLocale: 'en_US',
        locale: 'en_US',
      }),
    },
  });
};

// export custom render method(s) and other helpers:
export { initFbt, renderWithoutProviders as render };
