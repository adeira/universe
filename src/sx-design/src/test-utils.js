// @flow

import { type Element } from 'react';
import { init as FbtInit, IntlVariations as FbtIntlVariations } from 'fbt';
// eslint-disable-next-line import/no-extraneous-dependencies
import {
  render as renderWithoutProviders,
  type RenderOptionsWithoutCustomQueries,
} from '@testing-library/react';

import getFbtTranslationsForLocale from './getFbtTranslationsForLocale';
import type { SupportedLocales } from './constants';

const initFbt = () => {
  FbtInit({
    translations: getFbtTranslationsForLocale('en-US'),
    hooks: {
      getViewerContext: () => ({
        GENDER: FbtIntlVariations.GENDER_UNKNOWN,
        locale: 'en-US',
      }),
    },
  });
};

const renderWithProviders = (
  ui: Element<any>,
  {
    locale = 'en-US',
    ...renderOptions
  }: { +locale: SupportedLocales, ...RenderOptionsWithoutCustomQueries } = {},
): $FlowFixMe => {
  return renderWithoutProviders(ui, {
    wrapper: ({ children }) => {
      const SxDesignProvider = require('./SxDesignProvider').default;
      return <SxDesignProvider locale={locale}>{children}</SxDesignProvider>;
    },
    ...renderOptions,
  });
};

// re-export everything
// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';

// override render method
export { initFbt, renderWithProviders as render, renderWithoutProviders };
