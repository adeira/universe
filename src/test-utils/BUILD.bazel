load("//:bazel/macros/build_yarn_workspace.bzl", "build_yarn_workspace")
load("//:bazel/macros/test_yarn_workspace.bzl", "test_yarn_workspace")

build_yarn_workspace(
    name = "test-utils",
    package_name = "@adeira/test-utils",
    deps = [
        "@npm//graphql",
    ],
)

test_yarn_workspace(
    name = "test",
    deps = [
        "@npm//graphql",
    ],
)
