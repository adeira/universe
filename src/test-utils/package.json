{
  "name": "@adeira/test-utils",
  "description": "Set of tools for easier testing in JS.",
  "homepage": "https://github.com/adeira/universe/tree/master/src/test-utils",
  "license": "MIT",
  "private": false,
  "version": "0.6.0",
  "main": "src/index",
  "sideEffects": false,
  "dependencies": {
    "@adeira/js": "^1.3.0",
    "@babel/runtime": "^7.12.5"
  },
  "peerDependencies": {
    "graphql": "^14.6.0"
  }
}
