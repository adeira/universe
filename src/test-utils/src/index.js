// @flow

import generateTestsFromFixtures from './generateTestsFromFixtures';
import evaluateGraphQLResolver from './evaluateGraphQLResolver';

export { evaluateGraphQLResolver, generateTestsFromFixtures };
