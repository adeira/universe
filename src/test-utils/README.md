```text
yarn add @adeira/test-utils
```

Optionally, install TS types (Flow types are included by default):

```text
yarn add --dev @types/adeira__test-utils
```

# `generateTestsFromFixtures`

This function loads all the fixtures (their file content) from your folder and it snapshots the result of your callback operation. This is handy when you have several example files, you want to perform some operation on top of them and snapshot the results.

```js
import { generateTestsFromFixtures } from '@adeira/test-utils';

function operation(input) {
  return doSomethingWithTheInput(input);
}

generateTestsFromFixtures(`${__dirname}/__fixtures__`, operation);
```

There are two additional modes which allow you to control the fixtures behavior more precisely. There are based on the fixture name:

- `simpleExample.graphql` - fixture will snapshot as expected
- `simpleExample.only.graphql` - only these snapshots will run (for debug only)
- `simpleExample.error.graphql` - it's expected that fixture will throw an error (expected to fail tests)

The error file convention is required to prevent unintentional snapshots of exceptions.

# `evaluateGraphQLResolver`

```js
const fields = Location.getFields();
expect(
  evaluateGraphQLResolver(fields.countryFlagURL, {
    country: ' ... ', // test value
  }),
).toBe(' ... ');
```
