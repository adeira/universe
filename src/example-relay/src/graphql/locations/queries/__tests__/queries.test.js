// @flow

import { generateTestsFromFixtures } from '@adeira/test-utils';

import executeTestQuery from '../../../services/executeTestQuery';

function operation(input) {
  return executeTestQuery(input);
}
generateTestsFromFixtures(`${__dirname}/__fixtures__`, operation);
