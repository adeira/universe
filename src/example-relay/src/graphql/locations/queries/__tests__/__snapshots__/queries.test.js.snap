// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: locations.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
query locations {
  locations(first: 5) {
    edges {
      node {
        id
        name
        country {
          locationId
          name
        }
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "locations": {
      "edges": [
        {
          "node": {
            "id": "TG9jYXRpb246Q0FfVVM",
            "name": "California",
            "country": {
              "locationId": "US",
              "name": "United States"
            }
          }
        },
        {
          "node": {
            "id": "TG9jYXRpb246TllfVVM",
            "name": "New York",
            "country": {
              "locationId": "US",
              "name": "United States"
            }
          }
        },
        {
          "node": {
            "id": "TG9jYXRpb246aW5kZXBlbmRlbmNlLW5hdGlvbmFsLWhpc3RvcmljYWwtcGFyay1wYS11c2FfcG9p",
            "name": "Independence National Historical Park, PA, USA",
            "country": null
          }
        },
        {
          "node": {
            "id": "TG9jYXRpb246bWFuaGF0dGFuLWlzbGFuZF9wb2k",
            "name": "Manhattan Island",
            "country": null
          }
        },
        {
          "node": {
            "id": "TG9jYXRpb246bWV0cm9wb2xpdGFuLW11c2V1bS1vZi1hcnRfcG9p",
            "name": "Metropolitan Museum of Art",
            "country": null
          }
        }
      ]
    }
  }
}
`;
